source/connect/csot.txt

```diff
@@ -1,210 +1,218 @@
+Nested Timeout Blocks
+~~~~~~~~~~~~~~~~~~~~~
+
+When execution enters a nested ``timeout()`` block, the outer ``timeout()`` deadline is
+paused. When execution exits the nested ``timeout()`` block, the previous ``timeout()``
+deadline resumes.
 
-When execution enters a nested ``timeout`` block, the outer ``timeout`` deadline is paused.
-When execution exits the nested ``timeout`` block, the previous ``timeout`` deadline resumes.
 The following example shows how nested calls to the ``timeout()`` method interact:
 
 .. code-block:: python
 
    with pymongo.timeout(5):
-       coll.find_one()  # Uses the 5 second deadline.
+       collection.find_one()  # Uses the 5 second deadline
        with pymongo.timeout(3):
-           coll.find_one() # Uses the 3 second deadline.
-       coll.find_one()  # Uses the original 5 second deadline.
+           collection.find_one() # Uses the 3 second deadline
+       collection.find_one()  # Uses the original 5 second deadline
        with pymongo.timeout(10):
-           coll.find_one()  # Still uses the original 5 second deadline.
-       coll.find_one()  # Uses the original 5 second deadline.
-
-Timeout Errors
---------------
-
-When execution enters a ``~pymongo.timeout()`` ``with`` block, a deadline is set
-for the entire block. When that deadline is exceeded, any blocking {+driver-short+}
-operation will raise a timeout exception. For example:
-
-.. code-block:: python
+           collection.find_one()  # Still uses the original 5 second deadline
+       collection.find_one()  # Uses the original 5 second deadline
 
-   try:
-       with pymongo.timeout(10):
-           coll.insert_one({"name": "Nunu"})
-           time.sleep(10)
-           # The deadline has now expired, the next operation will raise
-           # a timeout exception.
-           coll.find_one({"name": "Nunu"})
-   except PyMongoError as exc:
-       if exc.timeout:
-           print(f"block timed out: {exc!r}")
-       else:
-           print(f"failed with non-timeout error: {exc!r}")
+Thread and Task Safety
+~~~~~~~~~~~~~~~~~~~~~~
 
-The ``pymongo.errors.PyMongoError.timeout`` property (added in PyMongo 4.2)
-is ``True`` when the error was caused by a timeout and ``False`` otherwise.
+The ``timeout()`` method is thread-safe. The timeout applies only to the current
+thread, and multiple threads can configure different timeouts in parallel.
 
-The timeoutMS URI Option
-------------------------
+The ``timeout()`` method is also ``asyncio``-safe. The timeout applies only to the
+current Task, and multiple Tasks can configure different timeouts at the same time.
 
-PyMongo 4.2 also added support for the ``timeoutMS`` URI and keyword argument to
-``~pymongo.mongo_client.MongoClient``. When this option is configured, the
-client automatically applies the timeout to each API call. For example:
+The following example shows how to use the ``timeout()`` method with
+`Motor <https://github.com/mongodb/motor>`__, the MongoDB
```

Comment:
Just a note - I think motor is being phased out (though idk how long that process is). So we may want to consider another example here. Probably a ticket for later but just wanted to call it out

Reply:
Tech reviewer - are there plans for this?

Reply:
@ShaneHarvey Can you comment on this?